kvpress/presses/expected_attention_press.py

```diff
@@ -67,45 +67,46 @@ def get_query_statistics(self, module: nn.Module, hidden_states: torch.Tensor):
         """
 
         bsz, q_len, _ = hidden_states.shape
-        n, d = module.config.num_attention_heads, module.head_dim
+        num_heads, head_dim = module.config.num_attention_heads, module.head_dim
 
         # Remove first hidden_states that likely contain outliers
         h = hidden_states[:, self.n_sink :]
 
-        if isinstance(module, (Qwen3Attention, Gemma3Attention)):
-            # Qwen and Gemma use QK norm, which is not compatible with ExpectedAttentionPress (for now)
-            raise NotImplementedError(f"ExpectedAttentionPress not yet implemented for {module.__class__}.")
-        elif isinstance(module, Phi3Attention):
-            Wq = module.qkv_proj.weight[: n * d]
+        if isinstance(module, Phi3Attention):
```

Comment:
Could this be extracted as a separate utils function?
E.g. `get_query_states(module, hissten_states, ...)`

Reply:
For snapkv, you could call the same function with `hidden_states[:, -window_size:]` as input?

> it would go to another file making the press non self-contained

I don't see an issue with that, given the helper function is rather slim.